### PR TITLE
Drug discovery generative models

### PR DESCRIPTION
**Pull request by agitter**
This isn't ready to be merged, but I wanted to share the Treat sections I'm working on.  Now that several of you are making edits (great!!!) we can work on small sub-sections to minimize the need to resolve conflicts when merging.

### CODE REVIEW

File: sections/05_treat.md [hunk 1]
```diff
@@ -257,11 +257,11 @@ study of BACE-1 inhibitors included only 1547 compounds
 this limited dataset, but overfitting was a challenge and the differences
 between random forests and a deep neural were negligible, especially in the
```
Comment (enricoferrero) [inline sections/05_treat.md]: You introduced the RF abbreviation before in this section (more generally, we need to agree whether to use the abbreviation or not and check when to first introduce it).

File: sections/05_treat.md [hunk 2]
```diff
@@ -257,11 +257,11 @@ study of BACE-1 inhibitors included only 1547 compounds
...
 classification setting.  Overfitting is still a problem in larger chemical
-screening datasets with tens or hundreds of thousands of compounds because the
-number of active compounds can be very small, on the order of 0.1% or 1% of all
-tested chemicals for a typical target  `TODO: verify those estimates`. This is
-consistent with the strong performance of low-parameter neural networks that
-emphasize compound-compound similarity, such as influence-relevance voter,
+screening datasets with tens or hundreds of thousands of compounds.  For a
```
Comment (enricoferrero) [inline sections/05_treat.md]: I think in the previous version it was perhaps clearer why overfitting is still problematic. Now you have two sentences that may or may not be related.

File: sections/05_treat.md [hunk 3]
```diff
@@ -373,14 +363,39 @@ development of unsupervised representation learning methods
...
+learning for drug discovery are *de novo* drug design and protein
+structure-based models.  Whereas the goal of virtual screening is to prediction
+the biochemical activity of hundreds of thousands to millions of chemicals, *de
+novo* drug design explores the much larger space of at least 10<sup>60</sup>
+organic molecules with drug-like properties that could be chemically synthesized
+[@doi:10.1002/wcms.1104].  Generative neural networks for drug design typically
+represent chemicals with the simplified molecular-input line-entry system
+(SMILES), a standard way string-based representation with characters that
+represent atoms, bonds, and rings [@tag:Segler2017_drug_design].
+Gómez-Bombarelli et al. designed a  SMILES-to-SMILES autoencoder to learn a
```
Comment (enricoferrero) [inline sections/05_treat.md]: (character) encoding issue?

File: sections/05_treat.md (hunk 1 above)
Comment (agitter) [inline sections/05_treat.md]: Added to #345 so we can decide to abbreviate or not consistently

File: sections/05_treat.md (hunk 2 above)
Comment (agitter) [inline sections/05_treat.md]: Reverted the phrasing

**Comment by agitter** [comment]
Thanks @enricoferrero, I fixed these problems.

**Comment by agitter** [comment]
I removed the WIP tag and am contracting the scope.  I'll make a new pull request for structure-based models later.

@cgreene or someone else, can you please review and merge?  @enricoferrero already reviewed once.

File: sections/05_treat.md [hunk 4]
```diff
@@ -208,54 +208,52 @@ which small molecules (also referred to as chemical compounds or ligands)
...
-dependent?` This task has been treated variously as a classification,
-regression, or
-ranking problem. In reality, it does not fit neatly into any of those
-categories.  An ideal algorithm will rank a sufficient number of active
-compounds before the inactives, but the rankings of actives relative to other
-actives and inactives are less important
+hits, abandoning lead compounds` This screening process can serve as one of the
+first steps in the long drug discovery pipeline, where novel chemicals are
+pursued for their ability to inhibit or enhance disease-relevant biological
+mechanisms.  The appeal of machine learning in this domain is the need to
```
Comment (agapow) [inline sections/05_treat.md]: Is the use of "machine learning" rather than deep learning deliberate here?

File: sections/05_treat.md [hunk 5]
```diff
@@ -208,54 +208,52 @@ which small molecules (also referred to as chemical compounds or ligands)
...
+improve the efficiency of the initial high-throughput screens such that
+sufficient candidate active compounds can be identified without exhaustively
+screening libraries of hundreds of thousands or millions of chemicals.  `TODO:
+is the sufficient number target dependent?` This task has been treated variously
+as a classification, regression, or ranking problem. In reality, it does not fit
+neatly into any of those categories.  An ideal algorithm will rank a sufficient
```
Comment (agapow) [inline sections/05_treat.md]: I'm a little uneasy about the use of "active" and "inactive" here (would have gone for "potential candidates" or "potentially useful") but will bow to the experts.

File: sections/05_treat.md [hunk 6]
```diff
@@ -208,54 +208,52 @@ which small molecules (also referred to as chemical compounds or ligands)
...
+summaries of atom-atom relationships in the molecular graph, and more
+sophisticated derived properties [@doi:10.1002/9783527628766].   Alternatively,
+chemicals can be characterized with the fingerprint bit vectors, textual
+strings, or novel learned representations described below. Neural networks have
+a long history in this domain [@tag:Baskin2015_drug_disc] `TODO: can add
+additional references besides this review`, and the 2012 Merck Molecular
+Activity Challenge on Kaggle generated substantial excitement about the
+potential for high-parameter deep learning approaches.  The winning submission
+was an ensemble that included a multitask multilayer perceptron network
+[@tag:Dahl2014_multi_qsar], and the Merck sponsors noted drastic improvements
```
Comment (agapow) [inline sections/05_treat.md]: Delete "Merck"

File: sections/05_treat.md [hunk 7]
```diff
@@ -208,54 +208,52 @@ which small molecules (also referred to as chemical compounds or ligands)
...
+approach, DeepTox [@tag:Mayr2016_deep_tox], was also the overall winner of
+another competition, the Toxicology in the 21st Century (Tox21) Data Challenge,
+it did not dominate alternative methods as thoroughly as in other domains.
+DeepTox was the top performer on 9 of 15 targets and highly competitive with the
+top performer on the others.  However, for many targets there was little
+separation between the top two or three methods.  A reliance on AUC ROC `TODO:
+define here?` for the evaluation (see Discussion) further hinders the ability to
+declare Tox21 as an outright success for deep learning.
 
 In retrospect, the nuanced Tox21 performance may be more reflective of the
```
Comment (agapow) [inline sections/05_treat.md]: Finding the "in retrospect" sentence a little opaque. The "Some of the ADMET datasets" sentence may also be missing a word.

File: sections/05_treat.md [hunk 8]
```diff
@@ -362,43 +350,67 @@ broad and diverse prediction tasks will be as important as novel neural network
 architectures and expanded public datasets to advance the state of the art and
```
Comment (agapow) [inline sections/05_treat.md]: Sentence gets a bit awkward around "expanded public datasets". Reword?

File: sections/05_treat.md [hunk 9]
```diff
@@ -362,43 +350,67 @@ broad and diverse prediction tasks will be as important as novel neural network
...
+SMILES strings produced by the autoencoder's decoder correspond to valid
+chemical structures.  More recent approaches train RNNs on compounds from ChEMBL
+[@doi:10.1093/nar/gkr777] to first obtain a generic generative model for
+drug-like compounds [@tag:Segler2017_drug_design
+@tag:Olivecrona2017_drug_design].  These generative models successfully learn
+the grammar of compound representations, with 94%
+[@tag:Olivecrona2017_drug_design] or nearly 98% [@tag:Segler2017_drug_design] of
+generated SMILES corresponding to valid molecular structures.  The initial RNN
+is then fine-tuned to generate molecules that are likely to be active against a
+specific target by either continuing RNN training on a small set of positive
```
Comment (agapow) [inline sections/05_treat.md]: delete "RNN" from "continuing RNN training"

File: sections/05_treat.md (hunk 4 above)
Comment (agitter) [inline sections/05_treat.md]: Yes, but I'm not too attached to it.  Deep learning in the domain is new(ish), but other machine learning is firmly established.  This first paragraph was intended to introduce the problem and what type of ML task it is before moving into deep learning.

Do you suggest changing it?

File: sections/05_treat.md (hunk 5 above)
Comment (agitter) [inline sections/05_treat.md]: For better or worse, this is the preferred language in the papers I've read.  Example from #141:

> The data points x are compounds tested in the experimental assay, while the labels y are binary
experimental outcomes for the experiment (e.g., active/inactive).

File: sections/05_treat.md (hunk 6 above)
Comment (agitter) [inline sections/05_treat.md]: I prefer to keep it following literature in this domain.  Subsequent papers (e.g. #251) refer to it using the Merck name:

> The first foray of deep learning into QSAR was the Merck challenge in 2012

File: sections/05_treat.md (hunk 7 above)
Comment (agitter) [inline sections/05_treat.md]: Tried to fix both.  Is this more readable?

> Some of the ADMET datasets showed superior performance in multi-task models with only 22
ADMET tasks compared to multi-task models with over 500 less-similar tasks.

File: sections/05_treat.md (hunk 8 above)
Comment (agitter) [inline sections/05_treat.md]: I cut "expanded public datasets", it wasn't critical to the point

File: sections/05_treat.md (hunk 9 above)
Comment (agitter) [inline sections/05_treat.md]: Done

File: sections/05_treat.md (hunk 4 above)
Comment (agapow) [inline sections/05_treat.md]: It sticks out a bit to me. Tough call given the history. Leave it.

File: sections/05_treat.md (hunk 6 above)
Comment (agapow) [inline sections/05_treat.md]: I meant as regards "Merck sponsors" not contest name. Is it relevant who sponsors were?

Comment (agitter) [inline sections/05_treat.md]: Oh, I see.  My intention there was that the authors of `[@tag:Ma2015_qsar_merck]`, Merck employees, made that remark in their paper.  The quote seemed more credible coming from the domain experts who have been working with that dataset before the challenge.  Is there a better phrasing?

Comment (agapow) [inline sections/05_treat.md]: I think their identity is implied without explicitly naming them. But let's not waste time over this. Make a call.

Comment (agitter) [inline sections/05_treat.md]: I decided to cut it

**Comment by agitter** [comment]
@enricoferrero and @agapow reviewed so I'm merging this myself.  There will be an opportunity for more comments when I clean up the remaining TODOs in the next day or two.

**Comment by mrwns** [comment]
There are a couple of references and context I would like to add to that section. Should this be done via a Pull Request?

**Comment by agitter** [comment]
I started making local edits throughout this section and will try to push them soon as a new pull request.  Which parts do you want to change?  I want to minimize merge conflicts.
